scripts/cmake/kconfig.cmake

```
@@ -1,9 +1,70 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-# Kconfig targets
-
 include(${CMAKE_CURRENT_LIST_DIR}/defconfigs.cmake)
 
+
+           ### configure-time .config ###
+
+
+if(NOT INIT_CONFIG_found)
+set(INIT_CONFIG "initial.config" CACHE STRING "Initial .config file")
```

I think we may need more comments for this behavior, I did some test and fine this worked like when you copy an existing config file and name it to `initial.config`, then no `-DINIT_CONFIG` is needed?
```suggestion
set(INIT_CONFIG "local.config" CACHE STRING "Default kconfig local.config")
```
I think a better name is needed, or `old.config` here?